src/cpp/rtps/builtin/BuiltinProtocols.cpp

```diff
@@ -129,6 +126,13 @@ bool BuiltinProtocols::initBuiltinProtocols(
         return false;
     }
 
+    {
+        std::unique_lock<std::recursive_mutex> lock(*mp_PDP->getMutex());
+        m_DiscoveryServers = m_att.discovery_config.m_DiscoveryServers;
+    }
+
+    transform_server_remote_locators(p_part->network_factory());
```

Comment:
:man_facepalming: I thought of running the Discovery Server tests locally and then I forgot. I think that the best solution right now is to not take the mutex. Although I would suggest leaving a `TODO` comment stating this situation in case in the future needs to be refactored.

Reply:
I agree